AppSettingsManagerAPI/AppSettingsManagerApi.Domain/Exceptions/InsufficientPermissionException.cs

```diff
@@ -0,0 +1,8 @@
+namespace AppSettingsManagerApi.Domain.Exceptions;
+
+public class InsufficientPermissionException : Exception
```

Comment:
I like the custom exceptions. Good to make these common issues reusable and have a useful name on an exception instead of just 'exception'